pandas/core/groupby/generic.py

```diff
@@ -893,6 +894,9 @@ def fillna(
         return result
 
     @doc(Series.take.__doc__)
+    @deprecate_nonkeyword_arguments(
+        version="3.0", allowed_args=["self", "axis", "indices"]
+    )
```

Comment:
Having another look, looks like ` deprecate_nonkeyword_arguments` doesn't deprecate the argument, but makes it usable only via a keyword argument (e.g. `foo(1)` would produce the warning, but `foo(val=1)` wouldn't). If that's the case, doesn't seem like we've got a decorator to simply deprecate an argument. In that case, you'll have to produce it manually.